### PR TITLE
Feature/55 docs

### PR DESCRIPTION
**Pull request by TimPurdum**
1. Created a `docs` folder that contains a Jekyll site for publishing via GH Pages.
2. Added a few basic pages like `Getting Started` and `Authentication`.
3. Created a pipeline in the build of `dymaptic.GeoBlazor.Core` that will generate markdown files automatically from XML comments and add them to the `docs\pages\classes`.
4. Added lots of XML comments to public classes and properties.

### CODE REVIEW

**Review by seahro — APPROVED** [review]
Jekyl autogenerated pages look good

File: docs/pages/classes/Components [hunk 1]
```diff
@@ -0,0 +1,209 @@
+---
```
Comment (morehavoc) [inline docs/pages/classes/Components:1]: @TimPurdum All of these files (in `classes`) are auto-generated, correct? Do we really want to check them all in?

Comment (TimPurdum) [reply docs/pages/classes/Components:1]: They are auto-generated, but only by building `dymaptic.GeoBlazor.Core`. So we'd have to customize the GH pages build to generate them. I haven't gotten that far in my knowledge.

Comment (morehavoc) [reply docs/pages/classes/Components:1]: I see, okay.

**Pull request merged**